pinot-query-runtime/src/main/java/org/apache/pinot/query/service/server/QueryServer.java

```diff
@@ -184,4 +211,27 @@ public void cancel(Worker.CancelRequest request, StreamObserver<Worker.CancelRes
     // we always return completed even if cancel attempt fails, server will self clean up in this case.
     responseObserver.onCompleted();
   }
+
+  private SslContext buildSslContext(TlsConfig tlsConfig) {
+    LOGGER.info("Building gRPC SSL context");
+    SslContext sslContext = CLIENT_SSL_CONTEXTS_CACHE.computeIfAbsent(tlsConfig.hashCode(), tlsConfigHashCode -> {
```

Comment:
Just for my own understanding; Why do we need to use a map for this? Is it simply so in the rare case that we launch multiple `QueryServer` we don't duplicate the SSLContexts?